UnityProject/Assets/Scripts/Systems/Explosions/ExplosionNode.cs

```diff
@@ -42,65 +47,123 @@ public void Process()
 				return;
 			}
 
-			EnergyExpended = metaTileMap.ApplyDamage(v3int, Damagedealt,
-			MatrixManager.LocalToWorldInt(v3int, matrix.MatrixInfo), AttackType.Bomb);
+            if (isEMP)
+            {
+				foreach(var thing in matrix.Get<ObjectBehaviour>(v3int, true))
+                {
+					if (thing.TryGetComponent<ItemStorage>(out var storage))
+					{
+						foreach (var slot in storage.GetItemSlotTree())
+						{
+							EMPItem(slot.ItemObject,(int)Damagedealt);
+						}
+					}
 
-			if (Damagedealt > 100)
-			{
-				var Node = matrix.GetMetaDataNode(v3int);
-				if (Node != null)
-				{
-					foreach (var electricalData in Node.ElectricalData)
+					if (thing.TryGetComponent<ItemAttributesV2>(out var item))
 					{
-						electricalData.InData.DestroyThisPlease();
+						EMPItem(thing.gameObject, (int)Damagedealt);
 					}
 
-					SavedPipes.Clear();
-					SavedPipes.AddRange(Node.PipeData);
-					foreach (var Pipe in SavedPipes)
+					if (thing.TryGetComponent<DoorMasterController>(out var door))
 					{
-						Pipe.pipeData.DestroyThis();
+						door.TriggerEMP();
+					}
+
+					if (thing.TryGetComponent<APCPoweredDevice>(out var powered))
+					{
+						powered.TriggerEMP((int)Damagedealt);
 					}
```

Comment:
Make an Emp interface

Reply:
IEMPable?

Reply:
do we want all APCPoweredDevices to be EMPable? Maybe that's where we should handle it

Reply:
Can you EMP lights? Doors?